### PR TITLE
Include nn.Parameter in ModelSummary

### LINKED ISSUE #12736: ModelSummary ignores nn.Parameter
## 🐛 Bug

I don't know if this is a bug of a feature, but currently ModelSummary only reports member `nn.Module`s and ignores `nn.Parameter`s that are directly created. I believe these parameters are an important part (and sometimes, a big part in terms of the number of weights) of the model so that they should be reported.

cc @borda @kaushikb11 @awaelchli @rohitgr7

### PR DESCRIPTION
**Pull request by yeonju7kim**
## What does this PR do?
Fixes #12736.
Redo #13895.

When we defined torch.nn.parameter object in the model and then print ModelSummary for the model, the printed result doesn't show the parameter number of the torch.nn.parameter object.
Let me show you the example.

``` python
import torch

from pytorch_lightning import LightningModule
from pytorch_lightning.utilities.model_summary import ModelSummary

class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.param = torch.nn.Parameter(torch.rand(2, 2))  # 2 * 2 = 4
        self.layer = torch.nn.Linear(2, 2)  # 2 * 2 + 2 = 6


if __name__ == "__main__":
    model = BoringModel()
    summary = ModelSummary(model)
    print(summary)
```

```
  | Name  | Type   | Params
---------------------------------
0 | param | Parameter | 4                        <- I add this line. 
1 | layer | Linear    | 6          
---------------------------------
10        Trainable params
0         Non-trainable params
10        Total params
0.000     Total estimated model params size (MB)
```

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Yes! It's my first contribution. I welcome your feedback!

### CODE REVIEW

File: src/pytorch_lightning/utilities/model_summary/model_summary.py [hunk 1]
```diff
@@ -254,6 +255,18 @@ def summarize(self) -> Dict[str, LayerSummary]:
 
         return summary
 
+    def summarize_parameter(self) -> List[Tuple[str, int]]:
+        """Summarizes the number of parameters in the parameter that is specified as a parameter explicitly."""
+        summary: List[Tuple[str, int]] = []
+        for (name, p) in self._model.named_parameters():
+            if any(s in name for s in [".weight", ".bias"]):
+                continue
```
Comment (akihironitta) [inline src/pytorch_lightning/utilities/model_summary/model_summary.py:263]: Why does it need to skip a parameter if its name is in `[".weight", ".bias"]`? Also, I think assuming that parameter names are either of `[".weight", ".bias"]` doesn't look reasonable as they can basically be any strings, no?
